### PR TITLE
set reader spec config and add test

### PR DESCRIPTION
**Pull request by jvenant**
Setup the config of the recordReaderSpec

 Fixes : #10727

### CODE REVIEW

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskExecutor.java [hunk 1]
```diff
@@ -287,6 +287,9 @@ protected SegmentGenerationTaskSpec generateTaskSpec(Map<String, String> taskCon
       recordReaderSpec.setDataFormat(taskConfigs.get(BatchConfigProperties.INPUT_FORMAT));
       recordReaderSpec.setClassName(taskConfigs.get(BatchConfigProperties.RECORD_READER_CLASS));
       recordReaderSpec.setConfigClassName(taskConfigs.get(BatchConfigProperties.RECORD_READER_CONFIG_CLASS));
+      recordReaderSpec.setConfigs(
```
Comment (snleee) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskExecutor.java:290]: Have we set up the code style? If not, let's set up the pinot style in IDE and apply the code style change.

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/test/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGeneratorTest.java [hunk 2]
```diff
@@ -78,4 +86,52 @@ public void testRealCluster()
     ControllerTest.sendPostRequest(_controllerRequestURLBuilder.forClusterConfigs(), request);
     Assert.assertEquals(_generator.getNumConcurrentTasksPerInstance(), 1);
   }
+
+  @Test
+  public void testGenerateTaskSpec() throws Exception {
+    URL resourcesLoc = SegmentGenerationAndPushTaskGeneratorTest.class.getClassLoader().getResource(".");
+    SegmentGenerationAndPushTaskExecutor executor = new SegmentGenerationAndPushTaskExecutor();
+    Schema schema = new Schema.SchemaBuilder().build();
+    // FieldUtils.writeField(executor, "_pinotTaskConfig", new PinotTaskConfig("",
```
Comment (snleee) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/test/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGeneratorTest.java:95]: (nit) Can we remove this line?

**Review by snleee — COMMENTED** [review]
Thank you for the contribution!

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskExecutor.java (hunk 1 above)
Comment (jvenant) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskExecutor.java]: Fixed in bb90840fffa9e1f3d2f03af2f8270127bfe91787

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/test/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGeneratorTest.java (hunk 2 above)
Comment (jvenant) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/test/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGeneratorTest.java]: Fixed in bb90840fffa9e1f3d2f03af2f8270127bfe91787

**Comment by jvenant** [comment]
Any feedback on this ?

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**